### PR TITLE
Dockerfile 

### PR DESCRIPTION
**Pull request by miguel-merlin**
#11 #13 
@ezrizhu We had to build the application before running. Now I am getting a connection refused error
```
Caused by: org.postgresql.util.PSQLException: Connection to localhost:5432 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.
```

Do you think it's something related to the docker-compose? This is how I am running the application
``` 
docker build -t blueprint_admin_backend .
docker-compose up -d --build 
docker run blueprint_admin_backend
```

### CODE REVIEW

**Comment by ezrizhu** [comment]
we can either
* use the -x test to skip the testing stage
* make the db server during build process

**Comment by miguel-merlin** [comment]
image
Works now

**Comment by ezrizhu** [comment]
```
λ docker build -t test .
[+] Building 92.2s (11/11) FINISHED                                                                                                                                                                                     docker:default
 => [internal] load build definition from Dockerfile                                                                                                                                                                              0.1s
 => => transferring dockerfile: 266B                                                                                                                                                                                              0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                 0.1s
 => => transferring context: 2B                                                                                                                                                                                                   0.0s
 => [internal] load metadata for docker.io/library/openjdk:17-slim                                                                                                                                                                7.1s
 => [internal] load build context                                                                                                                                                                                                 0.0s
 => => transferring context: 53.53kB                                                                                                                                                                                              0.0s
 => [1/7] FROM docker.io/library/openjdk:17-slim@sha256:aaa3b3cb27e3e520b8f116863d0580c438ed55ecfa0bc126b41f68c3f62f9774                                                                                                         12.5s
 => => resolve docker.io/library/openjdk:17-slim@sha256:aaa3b3cb27e3e520b8f116863d0580c438ed55ecfa0bc126b41f68c3f62f9774                                                                                                          0.0s
 => => sha256:44d3aa8d076675d49d85180b0ced9daef210fe4fdff4bdbb422b9cf384e591d0 1.58MB / 1.58MB                                                                                                                                    0.1s
 => => sha256:6ce99fdf16e86bd02f6ad66a0e1334878528b5a4b5487850a76e0c08a7a27d56 187.90MB / 187.90MB                                                                                                                                7.2s
 => => sha256:aaa3b3cb27e3e520b8f116863d0580c438ed55ecfa0bc126b41f68c3f62f9774 547B / 547B                                                                                                                                        0.0s
 => => sha256:779635c0c3d23cc8dbab2d8c1ee4cf2a9202e198dfc8f4c0b279824d9b8e0f22 953B / 953B                                                                                                                                        0.0s
 => => sha256:37cb44321d0423bc57266a3bff658daf00478e4cdf2d3b8091f785310534256d 4.80kB / 4.80kB                                                                                                                                    0.0s
 => => sha256:1fe172e4850f03bb45d41a20174112bc119fbfec42a650edbbd8491aee32e3c3 31.38MB / 31.38MB                                                                                                                                  1.9s
 => => extracting sha256:1fe172e4850f03bb45d41a20174112bc119fbfec42a650edbbd8491aee32e3c3                                                                                                                                         4.2s
 => => extracting sha256:44d3aa8d076675d49d85180b0ced9daef210fe4fdff4bdbb422b9cf384e591d0                                                                                                                                         0.3s
 => => extracting sha256:6ce99fdf16e86bd02f6ad66a0e1334878528b5a4b5487850a76e0c08a7a27d56                                                                                                                                         4.9s
 => [2/7] WORKDIR /app                                                                                                                                                                                                            6.6s
 => [3/7] COPY gradlew /app/                                                                                                                                                                                                      0.0s
 => [4/7] COPY gradle /app/gradle                                                                                                                                                                                                 0.0s
 => [5/7] COPY build.gradle /app/                                                                                                                                                                                                 0.0s
 => [6/7] RUN ./gradlew --no-daemon dependencies                                                                                                                                                                                 65.8s
 => ERROR [7/7] COPY build/libs/*.jar app.jar                                                                                                                                                                                     0.0s
------
 > [7/7] COPY build/libs/*.jar app.jar:
------
Dockerfile:11
--------------------
   9 |     RUN ./gradlew --no-daemon dependencies
  10 |
  11 | >>> COPY build/libs/*.jar app.jar
  12 |
  13 |     EXPOSE 8080
--------------------
ERROR: failed to solve: lstat /var/lib/docker/tmp/buildkit-mount328805101/build/libs: no such file or directory
```

**Comment by miguel-merlin** [comment]
@ezrizhu Can you re-review this PR. To run the prod version of the API 
```

mv  docker-compose.yaml docker-compose.dev.yaml
mv docker-compose.prod.yaml docker-compose.yaml
./gradlew build
docker-compose up --build -d
```
This will create the postgres db and bootup the service.

**Comment by ezrizhu** [comment]
@miguel-merlin just making sure I got it right, we're building the jar outside of docker, and just copying it into the docker container on build.
Why aren't we building it in the container buildtime?

File: Dockerfile
Comment (ezrizhu) [inline Dockerfile:1]: why are we building it outside of the docker buildtime

**Review by ezrizhu — CHANGES_REQUESTED** [review]

**Comment by miguel-merlin** [comment]
@ezrizhu I added a `build` stage in the dockerfile that creates the `jar` inside the container.

**Comment by miguel-merlin** [comment]
@ezrizhu I fixed all the merge conflicts. I will merge this

**Comment by ezrizhu** [comment]
looks good to me, tysm

**Pull request merged**